### DISCUSSION TITLE
Ignore height removes prerelease and incrementing patch version number

### DISCUSSION THREAD

**Discussion by kegesch** [adamralph/minver#752] [Q&A] [answered]
The documentation about `Height` describes it at as a number (= the amount of commits since the last found tag) that is concatted to the prerelease version number.
However, if you disable that behaviour (e.g. because you want to version multiple artefacts in a monorepo, as the the docs describe), it not only disables the height, but also the addition if the prerelease.

Example, the last tag is `0.2.0` and i have one commit on top (height = 1). 
`minver` results in:
``` 
MinVer: Using { Commit: 40720aa, Tag: '0.2.0', Version: 0.2.0, Height: 1 }.
MinVer: Calculated version 0.2.1-alpha.0.1.
0.2.1-alpha.0.1
```
`minver -i` results in:
```
MinVer: Using { Commit: 40720aa, Tag: '0.2.0', Version: 0.2.0, Height: 1 }.
MinVer: Calculated version 0.2.0.
0.2.0
```
Isn't the exepected result `0.2.1-alpha`?

**Comment by adamralph** [comment, marked as answer]
@kegesch the expected result is indeed `0.2.0`.

(BTW—in `0.2.1-alpha.0.1`, only the last `1` is the height. If you remove the height from that version, you are left with `0.2.1-alpha.0`.)

The option to ignore height means "use the tag as if it is on the current commit". I.e. as if it has no height.

You are interpreting the option as meaning "use the tag normally, but do not include its height in the calculated version".

I can see why you might argue that your interpretation may also be described as "ignoring height". However, I believe the current behaviour is correct.

For example:

- Artifact `B` references artifact `A`.
- They each use a version prefix of `B` and `A` respectively.
- I tag the latest commit with `A0.2.0`
- `A` now builds with a version of `0.2.0`.
- `B` now builds with a reference to `A 0.2.0`.
- I intend to work only on `B` now, so I set the versioning of `A` to ignore height, because I don't want commits which only affect `B` to affect the version of `A`.
  - _(This is the entire reason that the option to ignore height was added to MinVer.)_
- I make commits which only affect `B`.
- With the current behaviour:
  - `A` continues to build with a version of `0.2.0`.
  - `B` continues to build with a reference to `A 0.2.0`.
  - That's what I want, because `A` has not changed.
- With the behaviour you are expecting:
  - `A` would now build with a version of `0.2.1-alpha.0`.
  - `B` would now build with a reference to `A 0.2.1-alpha.0`.
  - I don't want that, since `A` has not changed.

I hope that makes sense.

**Reply by kegesch** [reply]
Your explanation make sense. A quick following up question. If you're "done" with task that work on `A` again and want to work on `B`, would you then re-enable height on `A` and ignore height on `B`?

**Reply by adamralph** [reply]
@kegesch I think you may have gotten `A` and `B` mixed up in your question but, essentially, yes, you are correct. Whenever you are working a given component, that component should _not_ be set to ignore height. And typically, I guess, yes, you _would_ ignore height on any components which are _not_ being worked on.

Incidentally, I guess this isn't the most elegant solution, but MinVer was not actually designed for this scenario (it's not something I do). Versioning separate components using prefixes is mentioned in the FAQ FWIW, since it can be done, and `MinVerIgnoreHeight` was added a concession to help anyone who might be doing that. Perhaps there are more elegant solutions available, but I probably won't be adding much more to MinVer for this purpose.

**Reply by kegesch** [reply]
Alright, thanks for your fast response!

**Reply by kegesch** [reply]
Wouldn't it quite easy to count the relevant height for an artifact with `git log <last-tag>.. <artifact's project folder>` ?
Would you be open for a PR?

**Reply by adamralph** [reply]
@kegesch that has been suggested before—please see the discussion in https://github.com/adamralph/minver/issues/616
